### ISSUE TITLE
openapi_context under xxx_get_subresource not working and wrong default summary

### ISSUE THREAD

**Issue by TheKassaK** [api-platform/core#4046]
**API Platform version(s) affected**: 2.6.2

**Description**  

I have a subresource : **/api/users/ {id}/basket**, so I retrieve the Basket resource belong to the user.

With this yml in users.yml : 

```
        api_users_basket_get_subresource:
            security: "is_granted('CAN_GET_SUB_RESOURCE_USER', _api_normalization_context['subresource_resources'])"
            normalization_context:
                without_permissions: true
                groups: [ "basket:read" ]
            openapi_context:
                summary: 'Retrieve the user Basket'
```
I cant change the summary, seems like openapi_context is ignored.

+

The default summary is wrong, for this route I have : **Retrieves a User resource.** instead of **Retrieves a Basket resource.**

**How to reproduce**  

use a subresource

**Comment by soyuka** [comment]
#4004 should've fixed that ?

**Comment by TheKassaK** [comment]
With this yml it's ok :

basket_get_subresource:
        security: "is_granted('CAN_GET_SUB_RESOURCE_USER', _api_normalization_context['subresource_resources'])"
        normalization_context:
            without_permissions: true
            groups: [ "basket:read" ]
        openapi_context:
            summary: "Retrieve User Basket resources"


So, **basket_get_subresource** instead of **api_users_basket_get_subresource**

If I check the route name in a subscriber for example with **$request->attributes->get('_route')**, I have **api_users_basket_get_subresource**

So it's weird, and the base problem is the wrong description for subressource operations.

**Comment by temp** [comment]
Same here. the "api" and $resource-prefix need to be removed to get this working, with is really counter-intuitive, because it looks like it isn't working at all...

**Comment by vincentchalamon** [comment]
Hi,

The `api` prefix is a known behavior in Subresources, like many other weird behaviors on Subresources...

We currently working on a refacto of the Subresources (check this out: https://github.com/api-platform/core/issues/2706 and https://github.com/api-platform/core/tree/main/docs/adr) to make them more flexible, more powerful and without those behaviors! :-)

Will be fixed in 2.7/3.0 with the new system.
